src/webgl/p5.Shader.js

```diff
@@ -1326,12 +1326,30 @@ p5.Shader = class {
         }
         break;
       case gl.SAMPLER_2D:
-        gl.activeTexture(gl.TEXTURE0 + uniform.samplerIndex);
-        uniform.texture =
-          data instanceof p5.Texture ? data : this._renderer.getTexture(data);
-        gl.uniform1i(location, uniform.samplerIndex);
-        if (uniform.texture.src.gifProperties) {
-          uniform.texture.src._animateGif(this._renderer._pInst);
+        if (typeof data == 'number') {
+          if (
+            data < gl.TEXTURE0 ||
+            data > gl.TEXTURE31 ||
+            data === Math.ceil(data)
```

Comment:
Does this need to be `!==`, since all the gl constants are ints?

Reply:
Fixed